assets/src/edit-story/components/library/panes/media/local/mediaPane.js

```diff
@@ -139,24 +142,54 @@ function MediaPane(props) {
         setMediaType,
         setSearchTerm,
         uploadVideoPoster,
+        optimizeVideo,
       };
     }
   );
 
   const { showSnackbar } = useSnackbar();
+  const enableMediaPickerVideoOptimization = useFeature(
+    'enableMediaPickerVideoOptimization'
+  );
 
   const {
+    allowedAllowedTranscodableTypes,
     allowedFileTypes,
     allowedMimeTypes: {
       image: allowedImageMimeTypes,
       video: allowedVideoMimeTypes,
     },
   } = useConfig();
 
-  const allowedMimeTypes = useMemo(
-    () => [...allowedImageMimeTypes, ...allowedVideoMimeTypes],
-    [allowedImageMimeTypes, allowedVideoMimeTypes]
-  );
+  const { isFeatureEnabled, isTranscodingEnabled } = useFFmpeg();
+
+  const allowedMimeTypes = useMemo(() => {
+    if (
+      enableMediaPickerVideoOptimization &&
+      isFeatureEnabled &&
+      isTranscodingEnabled
+    ) {
```

Comment:
If transcoding is supported, then add transcodable mime types to list of allowed types in the media picker.